fix: add missing namespace

adekbadek opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Removal of [this line](https://github.com/Automattic/newspack-plugin/pull/3951/files#diff-9a359e18b4a51063079daf9d0521c1cb0bee70f1077d98e3ff25b42c5c49092fL12) in #3951 causes email change to fail.  

Note: this was caught by E2E tests (running locally against `trunk`) 🎉

### How to test the changes in this Pull Request:

1. On `trunk`, complete the email change in My Account
2. Observe that on the last step (clicking the link in the confirmation email), there's a fatal error
3. Switch to this branch, observe no fatal error

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

adekbadek committed:
fix: add missing namespace

dkoo reviewed (changes requested):
Ah, I think we should just restore the `use` statements for the classes used here. Maybe moving the file to a different folder caused some different behavior. These are the ones I still see in use in this class:

```php

use Newspack\Reader_Activation;
use Newspack\Reader_Activation\ESP_Sync;
use Newspack\Stripe_Connection;
use Newspack\WooCommerce_Connection;
```

adekbadek committed:
fix: handle WP_Error in sync_email_change_with_esp

adekbadek commented:
I've added `use Newspack\Reader_Activation\ESP_Sync;` in 693cd949d258da4190ea318ee112f13b70550a97 - the file is already namespaced to `Newspack`, so the other `use` statements are not needed.

Also added error handling for `ESP_Sync::get_contact_data` result, which was missing.
